lib/release_notes_bot/persists.ex

```diff
@@ -68,18 +69,63 @@ defmodule ReleaseNotesBot.Persists do
       )
 
     if sanitizer.valid? do
-      case select_provider_to_persist(project_provider.persistence_provider_id, sanitizer.changes) do
+      case choose_action(sanitizer.changes, action, project_provider.persistence_provider_id) do
         {:ok, endpoint} ->
           {:ok, endpoint}
 
-        {:error, message} ->
-          {:error, message}
+        {:error, err} ->
+          {:error, err}
       end
     else
       {:error, "Invalid params.. Could not sanitize.."}
     end
   end
 
+  defp choose_action(sanitizer_changes, action, persistence_provider_id) do
+    case action do
+      "published" ->
+        case create(persistence_provider_id, sanitizer_changes) do
+          {:ok, res} ->
+            {:ok, res}
+
+          {:error, err} ->
+            {:error, err}
+        end
+
+      "edited" ->
+        case update(persistence_provider_id, sanitizer_changes) do
+          {:ok, res} ->
+            {:ok, res}
+
+          {:error, err} ->
+            {:error, err}
+        end
+
+      _ ->
+        {:error, "Invalid action"}
+    end
+  end
```

Comment:
We could alternatively write this as a pattern matched function instead of leveraging a simple case statement.

Each signature would capture the action name, for example 
```ex
choose_action(changes, "published", prov_id)`
``` 
and return the subsequently desired call. Otherwise we could fallback to a catchall like 
```ex
choose_action(_, _, _), do:   {:error, "Invalid action"}`
```
 From what Craig J told me, it takes advantage of compile time optimizations of the language.

Reply:
Is this really resolved? It's not necessary for merging, I'd say. But if we don't want to make the change immediately, let's preserve this conversation as unresolved for posterity

Reply:
Yessir - I went into this more in the next PR. Its not yet optimal - but can definitely refac more